File: Perez-Meneses-Juan-Carlos/Hilos/thread11_create.cpp
#include <iostream>
#include <thread>
#include <vector>

#define NUM_THREADS 5

void printHello(int threadId) {
    std::cout << "Hello World! It's me, thread #" << threadId << "!" << std::endl;
}

int main() {
    std::vector<std::thread> threads;
    for (int t = 0; t < NUM_THREADS; ++t) {
        std::cout << "In main: creating thread " << t << std::endl;
        threads.emplace_back(printHello, t);
    }

    for (auto &thread : threads) {
        thread.join();
    }

    return 0;
}

File: Perez-Meneses-Juan-Carlos/Hilos/thread11_create.py
import threading

NUM_THREADS = 5

def print_hello(thread_id):
    print(f"Hello World! It's me, thread #{thread_id}!")

threads = []
for t in range(NUM_THREADS):
    print(f"In main: creating thread {t}")
    thread = threading.Thread(target=print_hello, args=(t,))
    threads.append(thread)
    thread.start()

for thread in threads:
    thread.join()
